### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by danielcNet-bet**

### CODE REVIEW

File: core/logger/logger_test.go [hunk 1]
```diff
@@ -93,6 +94,7 @@ func TestLoggerReturnError(t *testing.T) {
 	goKitLogger := logger.NewLogger(loggerManager)
 	log := logger.SetLog(goKitLogger, loggerManager)
 	logErr := log.Logger.Log(params...)
+	fmt.Println(logErr)
```
Comment (kolbis) [inline core/logger/logger_test.go:97]: should remove printf
Add a test for when an error occurs and check the error you are producing.

File: core/logger/logger_manager.go [hunk 2]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
...
 // Panic print logs to all Loggers on panic level
 func (loggerManager loggerManager) Panic(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, PanicLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print PanicLoggerLevel log", logErr, log)
+			err = loggerManager.addError(err, logErr.Error())
```
Comment (kolbis) [inline core/logger/logger_manager.go:94]: err you pass is nil
you shluld have something like:

msg := fmt.Srintf("some message")
err = loggerManager.addError(logErr.Error(), msg)

You dont want to lose the cause of the error.
It is importnat to understand how errors lib work.

File: core/logger/logger_manager.go [hunk 3]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
...
 
 // Panic print logs to all Loggers on panic level
 func (loggerManager loggerManager) Panic(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, PanicLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print PanicLoggerLevel log", logErr, log)
```
Comment (kolbis) [inline core/logger/logger_manager.go:93]: no println

File: core/logger/logger_manager.go [hunk 4]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
...
 
 // Error print logs to all Loggers on error level
 func (loggerManager loggerManager) Error(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, ErrorLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print ErrorLoggerLevel log", logErr, log)
```
Comment (kolbis) [inline core/logger/logger_manager.go:80]: no println

err you pass is nil
you shluld have something like:

msg := fmt.Srintf("some message")
err = loggerManager.addError(logErr.Error(), msg)

You dont want to lose the cause of the error.
It is importnat to understand how errors lib work.

File: core/logger/logger_manager.go [hunk 5]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
...
 
 // Warn print logs to all Loggers on warn level
 func (loggerManager loggerManager) Warn(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, WarnLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print WarnLoggerLevel log", logErr, log)
```
Comment (kolbis) [inline core/logger/logger_manager.go:67]: no println

err you pass is nil
you shluld have something like:

msg := fmt.Srintf("some message")
err = loggerManager.addError(logErr.Error(), msg)

You dont want to lose the cause of the error.
It is importnat to understand how errors lib work.

File: core/logger/logger_manager.go [hunk 6]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
...
 
 // Info print logs to all Loggers on info level
 func (loggerManager loggerManager) Info(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, InfoLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print InfoLoggerLevel log", logErr, log)
```
Comment (kolbis) [inline core/logger/logger_manager.go:54]: no println

err you pass is nil
you shluld have something like:

msg := fmt.Srintf("some message")
err = loggerManager.addError(logErr.Error(), msg)

You dont want to lose the cause of the error.
It is importnat to understand how errors lib work.

File: core/logger/logger_manager.go [hunk 7]
```diff
@@ -33,80 +34,72 @@ func NewLoggerManager(loggers []Logger) Manager {
 
 // Debug print logs to all Loggers on debug level
 func (loggerManager loggerManager) Debug(ctx context.Context, message string, params ...interface{}) error {
-	var isError bool
+	var err error
 	for _, log := range loggerManager.Loggers {
 		logErr := log.Log(ctx, DebugLoggerLevel, message, params...)
 		if logErr != nil {
-			isError = true
 			fmt.Println("Cannot print DebugLoggerLevel log", logErr, log)
```
Comment (kolbis) [inline core/logger/logger_manager.go:41]: no println

err you pass is nil
you shluld have something like:

msg := fmt.Srintf("some message")
err = loggerManager.addError(logErr.Error(), msg)

You dont want to lose the cause of the error.
It is importnat to understand how errors lib work.

**Review by kolbis — CHANGES_REQUESTED** [review]

File: core/logger/logger_manager.go (hunk 2 above)
Comment (danielcNet-bet) [reply core/logger/logger_manager.go:88]: That exactly how its woks
we evaluate err var each time
if err return nil - no errors in all logs

File: core/logger/logger_manager.go (hunk 4 above)
Comment (danielcNet-bet) [reply core/logger/logger_manager.go]: That exactly how its woks
we evaluate err var each time
if err return nil - no errors in all logs

File: core/logger/logger_manager.go (hunk 6 above)
Comment (danielcNet-bet) [reply core/logger/logger_manager.go]: That exactly how its woks
we evaluate err var each time
if err return nil - no errors in all logs

File: core/logger/logger_manager.go (hunk 7 above)
Comment (danielcNet-bet) [reply core/logger/logger_manager.go]: That exactly how its woks
we evaluate err var each time
if err return nil - no errors in all logs

**Pull request closed without merging**